syntax = "proto3";
import "google/protobuf/empty.proto";

enum NotificationTypeEnum {
  email = 0;
  sms = 1;
  push = 2;
}



message SendNotificationRequest {
  string manager_id = 1;
  repeated string user_ids = 2;
  string notification_id = 3;
  string template_id = 4;
  string subject = 5;
  string text = 6;
  NotificationTypeEnum type = 7;
}

message CreateDelayedNotificationRequest {
  string manager_id = 1;
  repeated string user_ids = 2;
  string notification_id = 3;
  string template_id = 4;
  string subject = 5;
  string text = 6;
  repeated NotificationTypeEnum type = 7;
  int32 delay = 8;
}



message NotificationResponse {
  string task_id = 1;
}

service ManagerNotification {
  rpc SendNotificationToUsers (SendNotificationRequest) returns (google.protobuf.Empty) {}
  rpc CreateDelayedNotification (CreateDelayedNotificationRequest) returns (google.protobuf.Empty) {}
}










